vector-stores/spring-ai-gemfire/src/main/java/org/springframework/ai/vectorstore/GemFireVectorStore.java

```diff
@@ -0,0 +1,505 @@
+
+	private static final String DISTANCE_METADATA_FIELD_NAME = "distance";
+
+	private static final String EMBEDDINGS = "/embeddings";
+
+	private final WebClient client;
+
+	private final EmbeddingClient embeddingClient;
+
+	private final int topKPerBucket;
+
+	private final String documentField;
+
+	public static final class GemFireVectorStoreConfig {
+
+		private final WebClient client;
+
+		private final int topKPerBucket;
+
+		private final String documentField;
+
+		public static Builder builder() {
+			return new Builder();
+		}
+
+		private GemFireVectorStoreConfig(Builder builder) {
+			String base = UriComponentsBuilder
+				.fromUriString("http{ssl}://{host}:{port}/gemfire-vectordb/v1/indexes/{index}")
+				.build(builder.sslEnabled ? "s" : "", builder.host, builder.port, builder.index)
+				.toString();
+
+			this.client = WebClient.create(base);
+			this.topKPerBucket = builder.topKPerBucket;
+			this.documentField = builder.documentField;
+		}
+
+		public static class Builder {
+
+			private String host;
+
+			private int port = DEFAULT_PORT;
+
+			private boolean sslEnabled;
+
+			private long connectionTimeout;
+
+			private long requestTimeout;
+
+			private String index;
+
+			private int topKPerBucket = DEFAULT_TOP_K_PER_BUCKET;
+
+			private String documentField = DEFAULT_DOCUMENT_FIELD;
+
+			public Builder withHost(String host) {
+				Assert.hasText(host, "host must have a value");
+				this.host = host;
+				return this;
+			}
+
+			public Builder withPort(int port) {
+				Assert.isTrue(port > 0, "port must be postive");
+				this.port = port;
+				return this;
+			}
+
+			public Builder withSslEnabled(boolean sslEnabled) {
+				this.sslEnabled = sslEnabled;
+				return this;
+			}
+
+			public Builder withConnectionTimeout(long timeout) {
+				Assert.isTrue(timeout >= 0, "timeout must be >= 0");
+				this.connectionTimeout = timeout;
+				return this;
+			}
+
+			public Builder withRequestTimeout(long timeout) {
+				Assert.isTrue(timeout >= 0, "timeout must be >= 0");
+				this.requestTimeout = timeout;
+				return this;
+			}
+
+			public Builder withIndex(String index) {
+				Assert.hasText(index, "index must have a value");
+				this.index = index;
+				return this;
+			}
+
+			public Builder withTopKPerBucket(int topKPerBucket) {
+				Assert.isTrue(topKPerBucket > 0, "topKPerBucket must be positive");
+				this.topKPerBucket = topKPerBucket;
+				return this;
+			}
+
+			public Builder withDocumentField(String documentField) {
+				Assert.hasText(documentField, "documentField must have a value");
+				this.documentField = documentField;
+				return this;
+			}
+
+			public GemFireVectorStoreConfig build() {
+				return new GemFireVectorStoreConfig(this);
+			}
+
+		}
+
+	}
+
+	private static final int DEFAULT_PORT = 9090;
+
+	private static final int DEFAULT_TOP_K_PER_BUCKET = 10;
+
+	private static final String DEFAULT_DOCUMENT_FIELD = "document";
+
+	public static final String INDEX_NAME = "spring-ai-index";
```

Comment:
Is this a default value?  Or is this the only name we accept and only expect to create 1 index?

Reply:
INDEX_NAME is not the default value but DEFAULT_DOCUMENT_FIELD this is a default value.